### DISCUSSION TITLE
mcux usb: should `usb_dc_ep_disable()` mark endpoint as unoccupied?

### DISCUSSION THREAD

**Discussion by mparanjpe** [zephyrproject-rtos/zephyr#51685] [Q&A] [answered]
In `drivers/usb/device/usb_dc_mcux.c`, the function `usb_dc_ep_disable()` currently marks the endpoint as disabled [here](https://github.com/zephyrproject-rtos/zephyr/blob/4a8073882b9441659967f57716824a108fe2ef7e/drivers/usb/device/usb_dc_mcux.c#L459). Shouldn't it also mark the endpoint as un-occupied? I'm thinking of this:
```
diff --git a/drivers/usb/device/usb_dc_mcux.c b/drivers/usb/device/usb_dc_mcux.c
index 1b94fb05d5..069c558727 100644
--- a/drivers/usb/device/usb_dc_mcux.c
+++ b/drivers/usb/device/usb_dc_mcux.c
@@ -457,6 +457,7 @@ int usb_dc_ep_disable(const uint8_t ep)
        }
 
        dev_state.eps[ep_abs_idx].ep_enabled = false;
+       dev_state.eps[ep_abs_idx].ep_occupied = false;
 
        return 0;
 }
```

**Comment by mmahadevan108** [comment]
@mparanjpe Thank you for pointing this. This change does make sense. Would you be able to submit a PR?

**Reply by mparanjpe** [reply, marked as answer]
@mmahadevan108 
> @mparanjpe Thank you for pointing this. This change does make sense. Would you be able to submit a PR?

:+1: there is already a PR: zephyrproject-rtos/zephyr#51726
